cognee/infrastructure/databases/graph/neptune_driver/adapter.py

```
@@ -0,0 +1,1445 @@
+            edges = [
+                (r["source_id"], r["target_id"], r["type"], r["properties"])
+                for r in raw_rels
+            ]
+
+            logger.debug(f"Retrieved subgraph with {len(nodes)} nodes and {len(edges)} edges for type {node_type.__name__}")
+            return (nodes, edges)
+
+        except Exception as e:
+            error_msg = format_neptune_error(e)
+            logger.error(f"Failed to get nodeset subgraph for type {node_type}: {error_msg}")
+            raise Exception(f"Failed to get nodeset subgraph: {error_msg}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)